src/hostname/hostnamed.c

```diff
@@ -250,8 +250,27 @@ static int get_firmware_vendor(char **ret) {
          return get_hardware_firmware_data("bios_vendor", ret);
 }
 
-static int get_firmware_date(char **ret) {
-         return get_hardware_firmware_data("bios_date", ret);
+static int get_firmware_date(uint64_t *ret) {
+         const char *k;
+         struct tm tm;
+         time_t v;
+         _cleanup_free_ char *bios_date = NULL;
+         int r;
+
+         r = get_hardware_firmware_data("bios_date", &bios_date);
+         if (r <= 0)
+                 return r;
+
+         k = strptime(bios_date, "%m/%d/%Y", &tm);
+         if (!k || *k)
+                 return r;
+
+        v = timegm(&tm);
+        if (v == (time_t) -1)
+                return r;
+
```

Comment:
timegm() and friends will silently mangle the data in the `struct tm` if not normalized (i.e. "feb 30" is specified or garbage like that). I think we should check for that, i.e. compare if the values we care about still match what we passed in. if we not then the input was apparently garbage, and we should return an error